mvx-esdt-safe/tests/mvx_esdt_safe_blackbox_setup.rs

```diff
@@ -241,37 +270,12 @@ impl MvxEsdtSafeTestState {
         &mut self,
         to: ManagedAddress<StaticApi>,
         opt_transfer_data: OptionalValueTransferDataTuple<StaticApi>,
-        opt_payment: Option<PaymentsVec<StaticApi>>,
+        payment: PaymentsVec<StaticApi>,
         expected_error_message: Option<&str>,
-    ) {
-        let tx = self
+        expected_log: Option<&str>,
+    ) -> Option<Log> {
```

Comment:
I don't think you need to return `Log`. I see you always check that log is not empty which is probably always true because (I believe) there will be logs for both successful and failed txs.